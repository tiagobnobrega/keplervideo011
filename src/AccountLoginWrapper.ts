import {
  AccountLoginServerComponent,
  IAccountLoginHandlerAsync,
  IAccountLoginServerAsync,
  IStatus,
  StatusType,
} from '@amzn/kepler-media-account-login';
import { store } from './store';

const accountLoginServerComponent = new AccountLoginServerComponent();

export class AccountLoginWrapper {
  m_accountLoginServer?: IAccountLoginServerAsync;

  async updateStatus(loginStatus: boolean) {
    console.info(
      'AccountLoginWrapper::updateStatus invoked with: ',
      loginStatus,
    );
    try {
      const statusBuilder = accountLoginServerComponent.makeStatusBuilder();
      statusBuilder.status(
        loginStatus ? StatusType.SIGNED_IN : StatusType.SIGNED_OUT,
      );
      const status = statusBuilder.build();
      await this.m_accountLoginServer?.updateStatus(status);
    } catch (err) {
      console.error(
        'AccountLoginWrapper::updateStatus Failed updating login status: ',
        err,
      );
    }
  }

  setupAccountLoginServer() {
    try {
      this.m_accountLoginServer = accountLoginServerComponent.getOrMakeServer();
    } catch (err) {
      this.m_accountLoginServer = undefined;
      console.error(
        'AccountLoginWrapper::setupAccountLoginServer failed creating account login server: ',
        err,
      );
      return;
    }

    const accountLoginHandler: IAccountLoginHandlerAsync = {
      handleReadStatus(): Promise<IStatus> {
        console.info('AccountLoginWrapper::handleReadStatus invoked.');
        try {
          // Example of returning an app Login status from Redux
          const statusBuilder = accountLoginServerComponent.makeStatusBuilder();
          const appLoginStatus = store.getState().settings.loginStatus;

          statusBuilder.status(
            appLoginStatus ? StatusType.SIGNED_IN : StatusType.SIGNED_OUT,
          );

          const status = statusBuilder.build();

          console.info('AccountLoginWrapper:: App read status: ', status);

          return Promise.resolve(status);
        } catch (err) {
          console.error(
            'AccountLoginWrapper::handleReadStatus exception: ',
            err,
          );
        }

        return Promise.reject(
          'AccountLoginWrapper::handleReadStatus creating login status failed.',
        );
      },
    };

    try {
      console.info('AccountLoginWrapper::set Handler in process.');
      this.m_accountLoginServer?.setHandler(accountLoginHandler);
    } catch (err) {
      console.error(
        'AccountLoginWrapper::setupAccountLoginServer failed to set handler: ',
        err,
      );
    }
  }

  onStart(): Promise<void> {
    console.info('AccountLoginWrapper onStart()');
    this.setupAccountLoginServer();
    return Promise.resolve();
  }

  onStop(): Promise<void> {
    // add stop service code here.
    console.info('AccountLoginWrapper onStop()');
    return Promise.resolve();
  }
}

export const AccountLoginWrapperInstance = new AccountLoginWrapper();

export const onStartService = (): Promise<void> => {
  return AccountLoginWrapperInstance.onStart();
};

export const onStopService = (): Promise<void> => {
  return AccountLoginWrapperInstance.onStop();
};
