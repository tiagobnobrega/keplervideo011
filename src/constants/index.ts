import { COLORS } from '../styles/Colors';

export enum DPADEventType {
  LEFT = 'left',
  RIGHT = 'right',
  UP = 'up',
  DOWN = 'down',
  PLAY = 'play',
  PAUSE = 'pause',
  PLAYPAUSE = 'playpause',
  SKIPBACKWARD = 'skip_backward',
  SKIPFORWARD = 'skip_forward',
  SELECT = 'select',
  MENU = 'menu',
  BACK = 'back',
  PAGE_UP = 'page_up',
  PAGE_DOWN = 'page_down',
  PAGE_LEFT = 'page_left',
  PAGE_RIGHT = 'page_right',
  INFO = 'info',
  MORE = 'more',
}

export const NetworkStatus = {
  CONNECTED: 'Connected',
  NOT_CONNECTED: 'Not Connected',
};

export const LoginStatus = {
  SIGNED_IN: true,
  SIGNED_OUT: false,
};

export const EVENT_KEY_DOWN = 0;
export const EVENT_KEY_UP = 1;

export const MAX_NARROW_SCREEN_WIDTH = 1300;

export const CAPTION_DISABLE_TEXT = 'Off';
export const CAPTION_DISABLE_ID = '-1';
export const L2_GRADIENT_COLORS = [
  `${COLORS.PALE_GRAY}99`,
  `${COLORS.MEDIUM_GRAY}99`,
  COLORS.CHARCOAL_GRAY,
];
export enum ReadyState {
  HAVE_NOTHING = 0,
  HAVE_METADATA = 1,
  HAVE_CURRENT_DATA = 2,
  HAVE_FUTURE_DATA = 3,
  HAVE_ENOUGH_DATA = 4,
}
