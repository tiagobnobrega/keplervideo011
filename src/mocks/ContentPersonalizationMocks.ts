import {
  ContentEntitlementBuilder,
  ContentIdBuilder,
  ContentIdNamespaces,
  ContentInteractionBuilder,
  ContentInteractionType,
  CustomerListEntryBuilder,
  CustomerListType,
  EntitlementType,
  IContentEntitlement,
  IContentId,
  IContentInteraction,
  ICustomerListEntry,
  IPlaybackEvent,
  IProfileId,
  PlaybackEventBuilder,
  PlaybackState,
  ProfileIdBuilder,
  ProfileIdNamespaces,
} from '@amzn/kepler-content-personalization';
import {
  ChannelDescriptorBuilder,
  IChannelDescriptor,
} from '@amzn/kepler-epg-provider';
import { VideoPlayer } from '@amzn/react-native-w3cmedia';

export function getMockContentID(id: string, namespace: string): IContentId {
  return new ContentIdBuilder().id(id).idNamespace(namespace).build();
}

export function getMockContentEntitlement(): IContentEntitlement {
  console.info('k_content_per: Building ContentEntitlement object in App');
  return new ContentEntitlementBuilder()
    .acquisitionTimestamp(new Date())
    .contentId(getMockContentID('testId', ContentIdNamespaces.NAMESPACE_CDF_ID))
    .entitlementType(EntitlementType.PURCHASE)
    .expirationTimestamp(new Date())
    .build();
}

export function getMockContentInteraction(
  interactionType: ContentInteractionType,
  content: IContentId,
): IContentInteraction {
  console.info('k_content_per: Building ContentInteraction object in App');
  return new ContentInteractionBuilder()
    .contentId(content)
    .interactionTimestamp(new Date())
    .contentInteractionType(interactionType)
    .profileId(getMockProfileId())
    .build();
}

export function getMockCustomerListEntry(
  listType: CustomerListType,
  content: IContentId,
): ICustomerListEntry {
  console.info('k_content_per: Building CustomerListEntry object in App');
  return new CustomerListEntryBuilder()
    .contentId(
      content
        ? content
        : getMockContentID('testId', ContentIdNamespaces.NAMESPACE_CDF_ID),
    )
    .addedTimestamp(new Date())
    .build();
}

export function getMockProfileId(): IProfileId {
  return new ProfileIdBuilder()
    .id('testProfile')
    .idNamespace(ProfileIdNamespaces.NAMESPACE_APP_INTERNAL)
    .build();
}

function getMockChannelDescriptor(): IChannelDescriptor {
  return new ChannelDescriptorBuilder()
    .identifier('UniqueChannelIdentifier')
    .majorNumber(1)
    .minorNumber(0)
    .build();
}

export function getDefaultMockPlaybackEvent(): IPlaybackEvent {
  console.info('k_content_per: Building PlaybackEvent object in App');
  return new PlaybackEventBuilder()
    .channelDescriptor(getMockChannelDescriptor())
    .contentId(getMockContentID('testId', ContentIdNamespaces.NAMESPACE_CDF_ID))
    .creditsPositionMs(5000)
    .durationMs(50000)
    .playbackPositionMs(100)
    .playbackState(PlaybackState.PAUSED)
    .profileId(getMockProfileId())
    .eventTimestamp(new Date())
    .buildActiveEvent();
}

export function getMockPlaybackEventForVideo(
  videoRef: React.MutableRefObject<VideoPlayer | null>,
  title: string,
  playbackState: PlaybackState,
): IPlaybackEvent {
  if (!videoRef.current) {
    throw new Error('k_content_per: Video player is not initialized');
  }
  return (
    new PlaybackEventBuilder()
      .playbackState(playbackState)
      .profileId(getMockProfileId())
      .contentId(
        getMockContentID('title', ContentIdNamespaces.NAMESPACE_CDF_ID),
      )
      // Multiplying it by 1000 since the duration retrieved from the player is in seconds
      .durationMs(videoRef.current.duration * 1000)
      // Multiplying it by 1000 since the currentTime retrieved from the player is in seconds
      .playbackPositionMs(videoRef.current.currentTime * 1000)
      .buildActiveEvent()
  );
}
