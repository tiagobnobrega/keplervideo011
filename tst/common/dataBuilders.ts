import { RotatorData } from '../../src/components/rotator/type';
import { MovieGridData } from '../../src/types/MovieGridData';
import { TitleData } from '../../src/types/TitleData';

export class TitleDataBuilder {
  private id: string;
  private title: string = 'some title';
  private description: string = 'some description';
  private duration: number = 0;
  private thumbnail: string =
    'http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg';
  private posterUrl: string =
    'http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg';
  private videoUrl: string =
    'http://edge-vod-media.cdn01.net/encoded/0000169/0169313/video_1880k/T7J66Z106.mp4?source=firetv&channelID=1234';
  private categories: string[] = ['Costa Rica Islands'];
  private channelID: string = '1234';
  private format: string = 'some format';
  private uhd: boolean = false;
  private secure: boolean = false;
  private rentAmount: string = '10';

  constructor(id: string) {
    this.id = id;
  }

  setTitle(title: string): TitleDataBuilder {
    this.title = title;
    return this;
  }

  setDescription(description: string): TitleDataBuilder {
    this.description = description;
    return this;
  }

  setDuration(duration: number): TitleDataBuilder {
    this.duration = duration;
    return this;
  }

  setThumbnail(thumbnail: string): TitleDataBuilder {
    this.thumbnail = thumbnail;
    return this;
  }

  setPosterUrl(posterUrl: string): TitleDataBuilder {
    this.posterUrl = posterUrl;
    return this;
  }

  setVideoUrl(videoUrl: string): TitleDataBuilder {
    this.videoUrl = videoUrl;
    return this;
  }

  setCategories(categories: string[]): TitleDataBuilder {
    this.categories = categories;
    return this;
  }

  setChannelID(channelID: string): TitleDataBuilder {
    this.channelID = channelID;
    return this;
  }

  setFormat(format: string): TitleDataBuilder {
    this.format = format;
    return this;
  }

  setUhd(uhd: boolean): TitleDataBuilder {
    this.uhd = uhd;
    return this;
  }

  setSecure(secure: boolean): TitleDataBuilder {
    this.secure = secure;
    return this;
  }

  setRentAmount(rentAmount: string): TitleDataBuilder {
    this.rentAmount = rentAmount;
    return this;
  }

  build(): TitleData {
    return {
      id: this.id,
      title: this.title,
      description: this.description,
      duration: this.duration,
      thumbnail: this.thumbnail,
      posterUrl: this.posterUrl,
      videoUrl: this.videoUrl,
      categories: this.categories,
      channelID: this.channelID,
      mediaType: 'video',
      mediaSourceType: 'url',
      format: this.format,
      uhd: this.uhd,
      secure: this.secure,
      rentAmount: this.rentAmount,
    };
  }
}

export class MovieGridDataBuilder {
  private heading: string;
  private testID: string;
  private tileData: TitleData[] = [];

  constructor(heading: string, testID: string) {
    this.heading = heading;
    this.testID = testID;
  }

  addTile(tileData: TitleData): MovieGridDataBuilder {
    this.tileData.push(tileData);
    return this;
  }

  build(): MovieGridData {
    return {
      heading: this.heading,
      testID: this.testID,
      data: () => this.tileData,
    };
  }
}

export class RotatorDataBuilder {
  private id: string;
  private title: string = 'title';
  private description: string = 'description';
  private duration: string = '100';
  private thumbnailUrl: string =
    'http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg';
  private posterUrl: string =
    'http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg';
  private videoUrl: string =
    'http://edge-vod-media.cdn01.net/encoded/0000169/0169313/video_1880k/T7J66Z106.mp4?source=firetv&channelID=1234';
  private categories: string[] = ['Costa Rica Islands'];
  private channelID: string = '100';
  private rating: string = '20';
  private releaseDate: string = '1/1/1970';

  constructor(id: string) {
    this.id = id;
  }

  setTitle(title: string): RotatorDataBuilder {
    this.title = title;
    return this;
  }

  setDescription(description: string): RotatorDataBuilder {
    this.description = description;
    return this;
  }

  setDuration(duration: string): RotatorDataBuilder {
    this.duration = duration;
    return this;
  }

  setThumbnailUrl(thumbnailUrl: string): RotatorDataBuilder {
    this.thumbnailUrl = thumbnailUrl;
    return this;
  }

  setPosterUrl(posterUrl: string): RotatorDataBuilder {
    this.posterUrl = posterUrl;
    return this;
  }

  setVideoUrl(videoUrl: string): RotatorDataBuilder {
    this.videoUrl = videoUrl;
    return this;
  }

  setCategories(categories: string[]): RotatorDataBuilder {
    this.categories = categories;
    return this;
  }

  setChannelID(channelID: string): RotatorDataBuilder {
    this.channelID = channelID;
    return this;
  }

  setRating(rating: string): RotatorDataBuilder {
    this.rating = rating;
    return this;
  }

  setReleaseDate(releaseDate: string): RotatorDataBuilder {
    this.releaseDate = releaseDate;
    return this;
  }

  build(): RotatorData {
    return {
      id: this.id,
      title: this.title,
      description: this.description,
      duration: this.duration,
      thumbnailUrl: this.thumbnailUrl,
      posterUrl: this.posterUrl,
      videoUrl: this.videoUrl,
      categories: this.categories,
      channelID: this.channelID,
      rating: this.rating,
      releaseDate: this.releaseDate,
    };
  }
}
