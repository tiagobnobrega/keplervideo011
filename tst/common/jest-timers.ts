const tick = () => new Promise(res => setImmediate(res));

const advanceTimersByTime = async (time: number) =>
  jest.advanceTimersByTime(time) && (await tick());

const runOnlyPendingTimers = async () =>
  jest.runOnlyPendingTimers() && (await tick());

const runAllTimers = async () => jest.runAllTimers() && (await tick());

export { advanceTimersByTime, runAllTimers, runOnlyPendingTimers, tick };
