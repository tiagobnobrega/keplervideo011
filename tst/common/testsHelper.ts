// testHepler.ts
export const asMock = <T extends any = any, Y extends any[] = any[]>(
  obj: any,
): jest.Mock<T, Y> => obj as jest.Mock<T, Y>;
