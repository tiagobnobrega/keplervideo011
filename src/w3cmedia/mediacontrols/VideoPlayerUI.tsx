import {
  ContentPersonalizationServer,
  IPlaybackEvent,
  PlaybackState,
} from '@amzn/kepler-content-personalization';
import { HWEvent, useTVEventHandler } from '@amzn/react-native-kepler';
import { VideoPlayer } from '@amzn/react-native-w3cmedia';
import React, { useState } from 'react';
import { Platform, StyleSheet, View } from 'react-native';
import { isContentPersonalizationEnabled } from '../../config/AppConfig';
import { DPADEventType, EVENT_KEY_DOWN } from '../../constants';
import { getMockPlaybackEventForVideo } from '../../mocks/ContentPersonalizationMocks';
import { ControlBar } from './ControlBar';
import ControlBarMenu from './ControlBarMenu';
import { PlaybackControls } from './PlaybackControls';
import Header from './VideoPlayerHeader';
interface VideoPlayerUIProps {
  videoRef: React.MutableRefObject<VideoPlayer | null>;
  navigateBack: () => void;
  title: string;
  setCaptionStatus: (status: boolean) => void;
  setSelectedCaptionInVideoPlayer: (id: string) => void;
}

export const VideoPlayerUI = ({
  navigateBack,
  title,
  videoRef,
  setCaptionStatus,
  setSelectedCaptionInVideoPlayer,
}: VideoPlayerUIProps) => {
  const [captionMenuVisibility, setCaptionMenuVisibility] =
    useState<boolean>(false);

  const setSelectedCaptionInMenuBar = (id: string) => {
    setSelectedCaptionInVideoPlayer(id);
  };

  const [playerControlType, setPlayerControlType] = useState<string>('');
  const [isCaptionButtonFocused, setCaptionButtonFocused] = useState(false);

  useTVEventHandler((evt: HWEvent) => {
    if (!Platform.isTV) {
      return;
    }
    if (evt && evt.eventKeyAction === EVENT_KEY_DOWN) {
      let playbackEvent: IPlaybackEvent | undefined;
      setPlayerControlType(evt.eventType);

      if (videoRef.current) {
        try {
          if (evt.eventType === DPADEventType.PLAYPAUSE) {
            console.info('k_content_per: VideoPlayerUI : playpause');
            if (videoRef.current.paused) {
              playbackEvent = getMockPlaybackEventForVideo(
                videoRef,
                title,
                PlaybackState.PLAYING,
              );
            } else {
              playbackEvent = getMockPlaybackEventForVideo(
                videoRef,
                title,
                PlaybackState.PAUSED,
              );
            }
          } else if (evt.eventType === DPADEventType.PLAY) {
            console.info('k_content_per: VideoPlayerUI : play');
            playbackEvent = getMockPlaybackEventForVideo(
              videoRef,
              title,
              PlaybackState.PLAYING,
            );
          } else if (evt.eventType === DPADEventType.PAUSE) {
            console.info('k_content_per: VideoPlayerUI : pause');
            playbackEvent = getMockPlaybackEventForVideo(
              videoRef,
              title,
              PlaybackState.PAUSED,
            );
          } else if (evt.eventType === DPADEventType.SKIPFORWARD) {
            console.info('k_content_per: VideoPlayerUI : skip_forward');
            playbackEvent = getMockPlaybackEventForVideo(
              videoRef,
              title,
              PlaybackState.PLAYING,
            );
          } else if (evt.eventType === DPADEventType.SKIPBACKWARD) {
            console.info('k_content_per: VideoPlayerUI : skip_backward');
            playbackEvent = getMockPlaybackEventForVideo(
              videoRef,
              title,
              PlaybackState.PLAYING,
            );
          } else if (evt.eventType === DPADEventType.BACK) {
            if (captionMenuVisibility) {
              setCaptionMenuVisibility(false);
            }
            setCaptionStatus(captionMenuVisibility);
            setCaptionButtonFocused(captionMenuVisibility);
          }
        } catch (e) {
          console.error(`k_content_per: ${e}`);
        }
      }

      if (playbackEvent) {
        if (isContentPersonalizationEnabled()) {
          console.log(
            `k_content_per: useTvEventHandler: Reporting new playback event : ${JSON.stringify(
              playbackEvent,
            )}`,
          );
          ContentPersonalizationServer.reportNewPlaybackEvent(playbackEvent);
        }
      }
    }
  });

  const toggleCaption = () => {
    setCaptionMenuVisibility(!captionMenuVisibility);
  };

  return (
    <View style={styles.uiContainer} testID="video-player-ui-view">
      <View style={styles.ui}>
        <Header navigateBack={navigateBack} title={title} />
        <PlaybackControls
          videoRef={videoRef}
          playerControlType={playerControlType}
        />
        <ControlBar
          videoRef={videoRef}
          captions={toggleCaption}
          captionMenuVisibility={captionMenuVisibility}
          playerControlType={playerControlType}
          isCaptionButtonFocused={isCaptionButtonFocused}
        />
      </View>
      <ControlBarMenu
        captionMenuVisibility={captionMenuVisibility}
        videoRef={videoRef}
        setSelectedCaptionInMenuBar={setSelectedCaptionInMenuBar}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  ui: {
    height: '100%',
    width: '100%',
    justifyContent: 'space-between',
  },
  uiContainer: {
    position: 'absolute',
    zIndex: 5,
    height: '100%',
    width: '100%',
  },
});

export default VideoPlayerUI;
