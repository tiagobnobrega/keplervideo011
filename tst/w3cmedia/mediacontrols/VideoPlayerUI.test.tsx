import { useTVEventHandler } from '@amzn/react-native-kepler';
import { VideoPlayer } from '@amzn/react-native-w3cmedia';
import { render, waitFor } from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import { consoleInfoSpy } from '../../../jest.setup';
import { DPADEventType } from '../../../src/constants';
import { getMockPlaybackEventForVideo } from '../../../src/mocks/ContentPersonalizationMocks';
import VideoPlayerUI from '../../../src/w3cmedia/mediacontrols/VideoPlayerUI';
import { asMock } from '../../common/testsHelper';

jest.mock('@amzn/kepler-content-personalization', () => ({
  __esModule: true,
  ContentPersonalizationServer: jest.fn(),
}));
let videoRef: React.MutableRefObject<VideoPlayer | null> = {
  current: {
    setSurfaceHandle: jest.fn(),
    setCaptionViewHandle: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    textTracks: {
      length: 2,
      getTrackById: jest.fn(id => ({
        mode: id === '0' ? 'showing' : 'hidden',
      })),
    },
  } as unknown as VideoPlayer,
};
const setCaptionStatusMock = jest.fn();
const setSelectedCaptionInVideoPlayerMock = jest.fn();
const renderVideoPlayerUI = (
  videoRefParam?: React.MutableRefObject<VideoPlayer | null> | undefined,
) => {
  return (
    <VideoPlayerUI
      videoRef={videoRefParam || videoRef}
      title={'Beautiful Whale Tail Uvita Costa Rica'}
      navigateBack={jest.fn()}
      setCaptionStatus={setCaptionStatusMock}
      setSelectedCaptionInVideoPlayer={setSelectedCaptionInVideoPlayerMock}
    />
  );
};
describe('BufferingWindow Test Cases', () => {
  it('renders correctly', () => {
    const tree = render(renderVideoPlayerUI());
    expect(tree).toMatchSnapshot();
  });

  it('View main is present in component', async () => {
    const { queryByTestId } = render(renderVideoPlayerUI());
    const view = queryByTestId('video-player-ui-view');
    expect(view).toBeDefined();
  });

  it('SubView is present in component', async () => {
    const { queryByTestId } = render(renderVideoPlayerUI());
    const view = queryByTestId('video-player-ui-view-two');
    expect(view).toBeDefined();
  });

  it('PlaybackControls is present in component', async () => {
    const { queryByTestId } = render(renderVideoPlayerUI());
    const view = queryByTestId('player-btn-seek-backward');
    expect(view).toBeDefined();
  });

  it('ControlBar is present in component', async () => {
    const { queryByTestId } = render(renderVideoPlayerUI());
    const view = queryByTestId('control-bar');
    expect(view).toBeDefined();
  });

  it('ControlBarMenu is present in component', async () => {
    const { queryByTestId } = render(renderVideoPlayerUI());
    const view = queryByTestId('view-control-bar-menu');
    expect(view).toBeDefined();
  });
});

describe('PlayerScreen hooks', () => {
  let mockUseTVEventHandler: (arg0: {
    eventKeyAction: number;
    eventType: DPADEventType | 'error';
  }) => void;

  beforeEach(() => {
    jest
      .spyOn(require('react-native').Platform, 'isTV', 'get')
      .mockReturnValue(true);
    mockUseTVEventHandler = asMock(useTVEventHandler).mock.calls[0][0];
  });

  test('useTVEventHandler called with TV and DPADEventType.PLAYPAUSE with pause', async () => {
    videoRef = {
      current: {
        ...videoRef.current,
        paused: true,
      } as unknown as VideoPlayer,
    };
    render(renderVideoPlayerUI(videoRef));
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: DPADEventType.PLAYPAUSE,
    });

    expect(videoRef?.current?.paused).toBeTruthy();
    expect(getMockPlaybackEventForVideo).toBeDefined();
    expect(consoleInfoSpy).toHaveBeenCalledWith(
      'k_content_per: VideoPlayerUI : playpause',
    );
  });

  test('useTVEventHandler called with TV and DPADEventType.PLAYPAUSE without pause', async () => {
    videoRef = {
      current: {
        ...videoRef.current,
        paused: false,
      } as unknown as VideoPlayer,
    };
    render(renderVideoPlayerUI());
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: DPADEventType.PLAYPAUSE,
    });

    expect(videoRef?.current?.paused).not.toBeTruthy();
    expect(getMockPlaybackEventForVideo).toBeDefined();
    expect(consoleInfoSpy).toHaveBeenCalledWith(
      'k_content_per: VideoPlayerUI : playpause',
    );
  });

  test('useTVEventHandler called with TV and DPADEventType.PLAY', async () => {
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: DPADEventType.PLAY,
    });
    expect(videoRef?.current?.paused).not.toBeTruthy();
    expect(getMockPlaybackEventForVideo).toBeDefined();
    expect(consoleInfoSpy).toHaveBeenCalledWith(
      'k_content_per: VideoPlayerUI : play',
    );
  });

  test('useTVEventHandler called with TV and DPADEventType.PAUSE', async () => {
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: DPADEventType.PAUSE,
    });
    expect(videoRef?.current?.paused).not.toBeTruthy();
    expect(getMockPlaybackEventForVideo).toBeDefined();
    expect(consoleInfoSpy).toHaveBeenCalledWith(
      'k_content_per: VideoPlayerUI : pause',
    );
  });

  test('useTVEventHandler called with TV and DPADEventType.SKIPFORWARD', async () => {
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: DPADEventType.SKIPFORWARD,
    });
    expect(videoRef?.current?.paused).not.toBeTruthy();
    expect(getMockPlaybackEventForVideo).toBeDefined();
    expect(consoleInfoSpy).toHaveBeenCalledWith(
      'k_content_per: VideoPlayerUI : skip_forward',
    );
  });

  test('useTVEventHandler called with TV and DPADEventType.SKIPBACKWARD', async () => {
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: DPADEventType.SKIPBACKWARD,
    });
    expect(videoRef?.current?.paused).not.toBeTruthy();
    expect(getMockPlaybackEventForVideo).toBeDefined();
    expect(consoleInfoSpy).toHaveBeenCalledWith(
      'k_content_per: VideoPlayerUI : skip_backward',
    );
  });

  test('useTVEventHandler called with TV and DPADEventType.BACK', async () => {
    videoRef = {
      current: {
        ...videoRef.current,
        paused: false,
      } as unknown as VideoPlayer,
    };
    const { getByTestId } = render(renderVideoPlayerUI());
    const controlBar = getByTestId('control-bar');

    expect(controlBar.props.children.props.captionMenuVisibility).toBe(false);

    controlBar.props.children.props.onPress();
    expect(controlBar.props.children.props.captionMenuVisibility).toBe(true);
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: DPADEventType.BACK,
    });

    expect(setCaptionStatusMock).toHaveBeenCalled();
  });

  test('useTVEventHandler called with TV and error', async () => {
    mockUseTVEventHandler({
      eventKeyAction: 0,
      eventType: 'error',
    });
    await waitFor(() => {
      expect(console.error).toHaveBeenCalled();
    });
  });
});
