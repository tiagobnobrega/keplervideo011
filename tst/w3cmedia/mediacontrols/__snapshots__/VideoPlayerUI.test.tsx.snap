// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BufferingWindow Test Cases renders correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "position": "absolute",
      "width": "100%",
      "zIndex": 5,
    }
  }
  testID="video-player-ui-view"
>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 0.8,
          "paddingLeft": 20,
          "paddingRight": 40,
          "paddingVertical": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hasTVPreferredFocus={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "padding": 25,
            "width": 125,
          }
        }
      >
        <Text
          testID="header_back_icon"
        />
      </View>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 70,
            "fontWeight": "500",
            "textAlign": "right",
            "width": "90%",
          }
        }
        testID="video-player-header"
      >
        Beautiful Whale Tail Uvita Costa Rica
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "padding": 25,
          }
        }
        testID="player-btn-seek-backward"
      >
        <Text />
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "padding": 25,
          }
        }
        testID="player-btn-play-pause"
      >
        <Text />
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "padding": 25,
          }
        }
        testID="player-btn-seek-forward"
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
      testID="control-bar"
    />
  </View>
  <View
    style={
      Object {
        "bottom": 90,
        "position": "absolute",
        "right": 0,
      }
    }
  />
</View>
`;
